python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py

```diff
@@ -210,6 +215,7 @@ def predict(self, data, batch_size=32, quantize=False):
                 from bigdl.nano.utils.log4Error import invalidInputError
                 invalidInputError(False,
                                   "You must call fit or restore first before calling predict!")
+            data = data.dataset.tensors[0].numpy() if isinstance(data, DataLoader) else data
```

Comment:
dataset.tensor is an internal implementation for `TensorDataset` only. This could be not valid for other customized dataloader.

Reply:
I think the best way is to change `_pytorch_fashion_inference` function and make it support dataloader directly.